enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2546,19 +2553,55 @@ func runBazelWrapper() error {
 	}
 	metadataFlag := "--build_metadata=EXPLICIT_COMMAND_LINE=" + string(originalArgsJSON)
 
-	startupArgs, err := customBazelrcOptions(rootPath, workspacePath)
+	// Apply custom startup args for BB-defined configuration.
+	bbStartupArgs, err := customBazelrcOptions(rootPath, workspacePath)
 	if err != nil {
 		return err
 	}
 
-	bazelCmd := append([]string{bazelBin}, append(startupArgs, originalArgs...)...)
+	bazelCmd := append([]string{bazelBin}, append(bbStartupArgs, originalArgs...)...)
```

Comment:
nit: store this `append(bbStartupArgs, originalArgs...)` in a local var `bazelStartupOptions` and reuse it for `cacheStartupOptions` below?